Add / update some redirects for old URLs and unify samples

thaJeztah opened the pull request:
Cleaning up branches on my machine, and found this one

See individual commits for details

thaJeztah commented:
@usha-mandya ptal

usha-mandya commented on samples/django.md:
```diff
@@ -270,9 +272,9 @@ In this section, you set up the database connection for Django.
+- [Docker Compose overview](../compose/index.md)
+- [Install Docker Compose](../compose/install.md)
+- [Getting Started with Docker Compose](../compose/gettingstarted.md)
+- [Docker Compose Command line reference](../compose/reference/index.md)
```
The Compose Command line reference page appears as a floating page as it is not linked from the toc.

usha-mandya commented:
@thaJezta Thanks for the PR. Left a comment about adding the Compose CLI reference page to the repo. Looks good otherwise.

thaJeztah commented:
> The Compose Command line reference page appears as a floating page as it is not linked from the toc.

@usha-mandya good catch; I removed the old "index.md", and moved the "overview.md" to take its place, now the compose reference index is at https://docs.docker.com/compose/reference/, instead of https://docs.docker.com/compose/reference/overview/

usha-mandya commented:
That's a good idea.

usha-mandya reviewed (approved):
LGTM
